filestore/filestore.go

```diff
@@ -48,19 +44,36 @@ func (fs fileStore) Create(p Path) (File, error) {
 	if err == nil {
 		return nil, fmt.Errorf("file %s already exists", name)
 	}
-	return newFile(Path(name))
+	return newFile(Path(fs.base), p)
 }
 
-func (fs fileStore) Store(p Path, src File) error {
+func (fs fileStore) Store(p Path, src File) (Path, error) {
 	dest, err := fs.Create(p)
 	if err != nil {
-		return err
+		return Path(""), err
 	}
 	defer dest.Close()
 	_, err = io.Copy(dest, src)
-	return err
+	if err != nil {
+		return Path(""), err
+	}
+	return Path(fs.filename(p)), nil
 }
 
 func (fs fileStore) Delete(p Path) error {
-	return os.Remove(fs.filename(p))
+	for idx := range fs.base {
+		if p[idx] != fs.base[idx] {
+			return fmt.Errorf("invalid base path for '%s' (expecting '%s')", string(p), fs.base)
+		}
+	}
```

Comment:
There is a [`strings.HasPrefix`](https://golang.org/pkg/strings/#HasPrefix) that can do this.